numba/experimental/jitclass/base.py

```diff
@@ -167,8 +167,17 @@ def register_class_type(cls, spec, class_ctor, builder):
     builder: the internal jitclass builder
     """
     # Normalize spec
-    if isinstance(spec, Sequence):
+    if spec is None:
+        spec = OrderedDict()
+    elif isinstance(spec, Sequence):
         spec = OrderedDict(spec)
+
+    # Extend spec with class annotations.
+    if hasattr(cls, "__annotations__"):
+        for attr, py_type in cls.__annotations__.items():
```

Comment:
Should we consider all `__annotations__` from the `__mro__`?

Consider this:

```python

In [8]: class A:
   ...:     x : int
   ...:

In [9]: A.__annotations__
Out[9]: {'x': int}

In [10]: class B(A):
    ...:     pass
    ...:

In [11]: B.__annotations__   # B didn't add new __annotations__ and carries A.__annotations__
Out[11]: {'x': int}

In [12]: class C(A):
    ...:     y : int
    ...:

In [13]: C.__annotations__   # C has new __annotations__ which should be consider incremental
Out[13]: {'y': int}
```

Reply:
Yup! TIL that `__annotations__` doesn't include fields from parent classes.  Updated and added a test for this.